project1/src/main/java/uk/co/cpsd/javaproject1/SimulatorFrame.java

```diff
@@ -24,16 +26,24 @@ public SimulatorFrame() {
 
         // Set up simulation timer (1 tick per 500ms)
         timer = new Timer(1000, (ActionEvent e) -> {
-            world.tick();
-            worldPanel.repaint();
+            world.moveAnimals();        // Make goats move + eat + act
+            world.tick();               // Tick: grow grass, energy, reproduction
+            worldPanel.repaint();       // Show updates on the screen
+        
+            if (world.getSecondsElapsed() >= tickLimitSimulation) {
+                world.writeToCSV(world.getGoatHistory(), world.getGrassHistory());
```

Comment:
Not convinced that CSV is doing anything useful here.  Sure, you've added some functionality, but why do we need it?  If you're going to say "to draw graphs in Excel" then I'd say "do it here".